depthai_nodes/ml/messages/segmentation.py

```diff
@@ -48,3 +49,29 @@ def mask(self, value: NDArray[np.int16]):
         if np.any((value < -1)):
             raise ValueError("Mask must be an array of integers larger or equal to -1.")
         self._mask = value
+
+    def getVisualizationMessage(self) -> dai.ImgFrame:
+        """Returns the default visualization message for segmentation masks."""
+        img_frame = dai.ImgFrame()
+        mask = self._mask.copy()
+
+        unique_values = np.unique(mask[mask >= 0])
+        scaled_mask = np.zeros_like(mask, dtype=np.uint8)
+
+        if unique_values.size == 0:
+            return img_frame.setCvFrame(scaled_mask, dai.ImgFrame.Type.BGR888i)
+
+        min_val, max_val = unique_values.min(), unique_values.max()
+
+        if min_val == max_val:
+            scaled_mask = np.ones_like(mask, dtype=np.uint8) * 255
+        else:
+            scaled_mask = ((mask - min_val) / (max_val - min_val) * 255).astype(
+                np.uint8
+            )
+        scaled_mask[mask == -1] = 0
```

Comment:
The background class becomes the same as the first class?

Reply:
yes because there cant be negative values in the mask when doing color mapping. Later, when the colormapping is done, the colors for background are set to [0, 0, 0] in line 74: `colored_mask[mask == -1] = [0, 0, 0]`